libraries/I2S/src/I2S.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+  Based off of ArduinoCore-SAMD I2S interface.  Modified for the
+  ESP8266 by Earle F. Philhower, III <earlephilhower@yahoo.com>
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include <Arduino.h>
+#include "I2S.h"
+
+I2SClass::I2SClass(bool enableTransmit, bool enableRecv, bool driveClocks) {
+  _enableTx = enableTransmit;
+  _enableRx = enableRecv;
+  _driveClk = driveClocks;
+  _running = false;
+  _onTransmit = nullptr;
+  _onReceive = nullptr;
+  _havePeeked = 0;
+  _peekedData = 0;
+  _bps = 0;
+  _writtenHalf = false;
+}
+
+int I2SClass::begin(int mode, long sampleRate, int bitsPerSample) {
```

Comment:
`int mode` -> `i2s_mode_t mode` ?
Or is it that the goal is to maintain the minimum difference with upstream lib ?

Reply:
Good catch.  Upstream lib has `int` IIRC, but it should be the enum.  Will fix.